assets/marketplace/main.js

```diff
@@ -0,0 +1,113 @@
+let vscode;
+try {
+    vscode = acquireVsCodeApi();
+} catch (error) {
+    
+}
+
+const app = new Vue({
+    el: '#app',
+    data: {
+        message: 'Hello Vue!',
+        searchInput: '',
+        selectedModule: {},
+        selectedTag: "",
+        moduleName: '',
+        modules: [],
+        endpoint: document.getElementById('app').getAttribute('data-endpoint').includes("{{") ? "http://localhost:50227" : document.getElementById('app').getAttribute('data-endpoint'),
+        errorMessage: "",
+    },
+    created: async function () {
+        this.modules = await this.getModules();
+        console.log(this.modules )
+    },
+    methods: {
+        getModules: async function () {
+            return (await axios.get(`${this.endpoint}/api/v1/modules`)).data;
+        },
+        getModuleMetadata: async function (module) {
+            return (await axios.get(module.iotEdgeMetadataUrl)).data;
+        },
+        showModule: async function (module) {
+            // this.selectedModule.id = '';
+            this.errorMessage = "";
+            const metadata = await this.getModuleMetadata(module);
+            this.selectedModule = Object.assign({}, module);
+            this.selectedModule.metadata = metadata;
+            this.selectedTag = this.selectedModule.metadata.tagsOrDigests[0];
+        },
+        importModule: async function () {

```

Comment:
how about just collect the information, and post the content to vscode message handler. So all the validation could be done in the message handler. And the front end just collect the necessary information?